### PR TITLE
Add support for Disable (Manifest) stage

### LINKED ISSUE #82: JSON patch strategy is broken
When running a JSON patch I get
```
json: cannot unmarshal array into Go struct field PatchManifestRequest.patchManifest.patchBody of type map[string]interface {}
```
With the payload:
```json
[
    {
        "op": "replace",
        "path": "/spec/template/spec/containers/0/image",
        "value": "registry.sdi.dev.weu.azure.paas.cloudcenter.corp/test-dev/test:0.0.1"
    }
]
```
Looks like this is because we are assuming the patch body to be of type `map[string]interface{}` when it should be `json.RawMessage`.

### PR DESCRIPTION
**Pull request by billiford**
- Fixes #82 
- Adds support for Disable (Manifest) stage
- Sets the default propagation policy for delete to foreground (see comments in code)

Notes:
- Reasoning for changing the default propagation policy is because during Highlander strategy Spinnaker makes a request to delete resources, but does not pass any options regarding whether to cascade delete or not. 
- I addressed the JSON patch issue here because OSS Clouddriver is using JSON patch for Enable/Disable manifest. I had an incorrect assumption of what a JSON patch was, but added tests to validate that a JSON patch request (which is an array instead of an object) can be unmarshaled into our patch manifest struct.

### CODE REVIEW

File: internal/api/core/kubernetes/delete.go [hunk 1]
```diff
@@ -31,20 +31,21 @@ func (cc *Controller) Delete(c *gin.Context, dm DeleteManifestRequest) {
...
-
 	if dm.Options.GracePeriodSeconds != nil {
 		do.GracePeriodSeconds = dm.Options.GracePeriodSeconds
 	}
-	// There are two options that will set the deletion to cascade. Originally this was set when
-	// `options.cascading = true`, however in later APIs they have this set in
-	// `options.orphanDependants = false`.
+	// There are two options that will set the deletion to orphan dependants:
+	// `options.cascading = false` and `options.orphanDependants = true`. The first
+	// happens when running a Delete (Manifest) stage and the second
```
Comment (iteaguy) [inline internal/api/core/kubernetes/delete.go:39]: Change from:
happens when running a Delete (Manifest) stage
to:
happens when running a Delete (Manifest) stage runs or executes

File: internal/api/core/kubernetes/delete.go [hunk 2]
```diff
@@ -31,20 +31,21 @@ func (cc *Controller) Delete(c *gin.Context, dm DeleteManifestRequest) {
...
 	//
-	// Using a pointer should prevent the older API versions from falsly cascading
-	// when they don't intend to.
-	propagationPolicy := v1.DeletePropagationOrphan
-	if dm.Options.Cascading ||
-		(dm.Options.OrphanDependants != nil && !*dm.Options.OrphanDependants) {
-		propagationPolicy = v1.DeletePropagationForeground
+	// Default to use the Delete Propagation Foreground as our propagation
+	// policy, in case neither are passed in, then check if either are
+	// passed and sent the propagation policy accordingly.
```
Comment (iteaguy) [inline internal/api/core/kubernetes/delete.go:44]: Change 'sent' to 'send' ?

File: internal/api/core/kubernetes/disable.go [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+	Path  string `json:"path"`
+	Value string `json:"value,omitempty"`
+}
+
+// patch attaches or detaches a given load balancer based on the op passed in ("add" or "remove")
+// from a target and then patches the target's labels. It uses the JSON patch strategy to do so.
+//
+// For a pod, this looks like:
+// [
+//   {"op": "add", "path": "/metadata/labels/key1", "value": "value1"
```
Comment (iteaguy) [inline internal/api/core/kubernetes/disable.go:195]: Missing '}' at the end of the comment

File: internal/api/core/kubernetes/ops.go [hunk 4]
```diff
@@ -82,15 +93,15 @@ type PatchManifestRequestOptions struct {
 
 // why are artifacts commented out here? possibly causing the problem of artifacts not getting bound correctly
```
Comment (dmrogers7) [inline internal/api/core/kubernetes/ops.go:94]: This comment can be removed

**Pull request merged**